### ISSUE TITLE
Unable to authenticate

### ISSUE THREAD

**Issue by ianseyer** [mongoeye/mongoeye#5]
Attempting to connect to an external Mongo DB on OS X using `MongoEYE v0.2` from the Darwin build.

Have tried several permutations of the following:
`mongoeye <url> <db> -u <user> -p <password> --auth-db <auth_db>` to no avail.

It gives me:
```
MongoEYE v0.2 - MongoDB exploration tool

Connecting: Error

Database requires authentication.
Please enter credentials using flags or environment variables.
```

Attempting to remove the `--auth-db` flag produces:
```
MongoEYE v0.2 - MongoDB exploration tool

Connecting: Error

Failed to authenticate.
Please make sure you have entered the correct credentials.
```

Despite the exact same credentials working for `mongo` command from the same host.

My mongo logs output:
```
2017-07-25T16:00:50.879+0000 [conn4778621]  authenticate db: <auth_db> { authenticate: 1, nonce: "xxx", user: "<user>", key: "xxx" }
2017-07-25T16:00:50.939+0000 [conn4778621] Unauthorized not authorized on admin to execute command { listDatabases: 1 }
```

However, I _am_ able to execute `show databases` from the mongo REPL on `admin`.

**Comment by michaljurecko** [comment]
Hi, thank you for your interest in this project

In [issue-2](https://github.com/mongoeye/mongoeye/tree/issue-2) branch, 
I [removed](https://github.com/mongoeye/mongoeye/commit/884d85133240764b3a2d38eaa6a793a2d154df17) unnecessary connection checks that can cause problems (with insufficient permissions).

**Please try the modified code.**

You can compile your own binary (from [issue-2](https://github.com/mongoeye/mongoeye/tree/issue-2) branch)
or use the one I've prepared: [issue-2.zip](https://github.com/mongoeye/mongoeye/files/1176014/issue-2.zip)

**Comment by ianseyer** [comment]
Attempting the same command on the updated code base yields:

```
MongoEYE DEV - MongoDB exploration tool

Connecting: Error

Connection failed: no reachable servers.
```

**Comment by michaljurecko** [comment]
Did you not forget to enter the name of the collection?

`mongoeye <url> <db> -u <user> -p <password> --auth-db <auth_db>`

`mongoeye <url> <db> <COLLECTION> -u <user> -p <password> --auth-db <auth_db>`

Most likely some conditions had to be changed in your environment, 
the message you got is directly from the `mgo` library:
![image](https://user-images.githubusercontent.com/19371734/28641415-74f70502-724f-11e7-8e14-d1e81bd5a71a.png)

**Comment by michaljurecko** [comment]
Any change? Modified version, which I sent should work without problems.

**Comment by michaljurecko** [comment]
Closed due to inactivity.

Simplified authentication merged (#6) to master branch.
